README.md

```diff
@@ -22,137 +22,50 @@ Thimble requires a modern web browser, and we recommend using Mozilla Firefox or
+### Requirements
+
+- Node.js (version 4.6 or later) [[download](https://nodejs.org/en/download/)]
+- Vagrant (version 1.9 or later) [[download](https://www.vagrantup.com/downloads.html)]
+  - __Note:__ On Windows machines, you may need to restart your computer after installing Vagrant for it to be fully usable
+- Virtualbox (version 5.1 or later) [[download](https://www.virtualbox.org/wiki/Downloads)]
+
+The setup of Thimble can be divided into two distinct sections:
+
+### Editor
+
+To fully use Thimble locally, you need to first setup Brackets locally first. This can be done by following the steps outlined below:
+
+- Fork the [Brackets repository](https://github.com/mozilla/brackets) and then clone it to your local machine using `git clone --recursive https://github.com/<your_username>/brackets.git` (replace `<your_username>` with your Github username for the account you forked Brackets into)
+- In the cloned repository directory, run `npm install` to install the dependencies for Brackets
+- Run `npm run build` to create the built editor files that will be used by Thimble
+- Run `npm start` to start a server that will allow the editor to be accessed on [http://localhost:8000/src](http://localhost:8000/src)
+- You can find out more information about setting up Brackets locally by referring to the instructions [here](https://github.com/mozilla/brackets#how-to-setup-bramble-brackets-in-your-local-machine)
+
+### Thimble and Services
+
+Thimble interacts with the Publish API (source managed in [publish.webmaker.org](https://github.com/mozilla/publish.webmaker.org)) to store users, projects, files and other content as well as publish user projects.
+
+For authentication and user management, Thimble uses Webmaker OAuth which consists of the Webmaker ID System (source managed in [id.webmaker.org](htps://github.com/mozilla/id.webmaker.org)) and the Webmaker Login API (source managed in [login.webmaker.org](https://github.com/mozilla/login.webmaker.org)).
+
+All three services along with Thimble are bundled together using Git subtrees and run together using Vagrant.
+
+The first step is to fork and clone Thimble and navigate to the cloned directory in a terminal shell.
+
+For the first time, to start all dependent services and Thimble, simply run:
+```
+vagrant up
+```
+This process can take about 10-15 minutes as it needs to download all dependencies. Once you see a log that says `Express server listening on http://localhost:3500`, you can access Thimble on [http://localhost:3500](http://localhost:3500).
+
+You can now make changes to the Thimble source code on your system and they should be automatically reflected on [http://localhost:3500](http://localhost:3500).
+
+To stop running Thimble, simply press `Ctrl+C` twice.
 
-Run `grunt requirejs:dist` to regenerate the front-end `dist/` folder if you so desire (it's
-only necessary in production). See `Gruntfile.js` for details.
+To restart Thimble again, run:
+```
+vagrant reload --provision
```

Comment:
Reloading Vagrant still takes a couple of minutes to complete, it can be especially annoying if automatic reloading doesn't work and you're trying to quickly debug or test changes. You'll get bogged down quickly with wait times.